### PR TITLE
feat(service-portal): Add rights portal 

### PR DESCRIPTION
**Pull request by disaerna**
# Sjúkratryggingar / Icelandic health insurance - Réttindagátt / Rights Portal

## What

* New client fetching data for the rights portal
* Move other health insurance client under same folder and making a parent folder structur for all clients related to Sjúkratryggingar
* New domain for Rights Portal 
* Refactor Health module for Service Portal and adding Therapies alongside Aids and nutrition under "Health"

## Why

We are starting the project to move Réttindagátt / Rights portal to Mínar síður / Service portal (My pages).
First in will be Þjálfun / Therapies and Næring og Hjálpartæki / Aids and Nutrition.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/health/src/components/FootNote.tsx/helper.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          link: (str) =>
+            button(
+              str,
+              'https://assets.ctfassets.net/8k0h54kbe6bj/7DTXLMx7vNP3CIPlZgl1uE/a5c0fea1979db348a8377023f6e40044/Vinnureglur_vegna_grei__slu____ttt__ku_Sj__kratrygginga____sj__kra__j__lfun.pdf',
+            ),
+        }),
+      }
+      break
+    case PHYSIO_ACCIDENT_THERAPY:
+      //TODO: Get correct disclaimer for different therapies
```
Comment (thordurhhh) [inline libs/service-portal/health/src/components/FootNote.tsx/helper.tsx:39]: Did you get those disclaimers?

File: libs/service-portal/health/src/components/FootNote.tsx/helper.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        second: formatMessage(messages['speechDisclaimer2']),
+      }
+    case OCCUPATIONAL_THERAPY:
+      //TODO: Get correct disclaimer for different therapies
+      return {
+        first: formatMessage(messages['occupationalDisclaimer'], {
+          link: (str) =>
+            button(
+              str,
+              'https://island.is/greidsluthatttaka-vegna-heilbrigdisthjonustu',
```
Comment (thordurhhh) [inline libs/service-portal/health/src/components/FootNote.tsx/helper.tsx:67]: We should put these links in a translation string so they can be updated in the CMS when they change, and point to the english version for english.

File: libs/service-portal/health/src/components/FootNote.tsx/helper.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import {
+  OCCUPATIONAL_THERAPY,
+  PHYSIO_ACCIDENT_THERAPY,
+  PHYSIO_HOME_THERAPY,
+  PHYSIO_THERAPY,
+  SPEECH_THERAPY,
+} from '../../utils/constants'
+
+//TODO: Get correct paths from Sjúkratryggingar
+const button = (text: string[], to?: string) => (
```
Comment (thordurhhh) [inline libs/service-portal/health/src/components/FootNote.tsx/helper.tsx:13]: This is the same as the LinkButton.
```libs/service-portal/health/src/components/LinkButton/LinkButton.tsx```

File: libs/service-portal/health/src/screens/AidsAndNutrition/AidsAndNutrition.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        />
+      ),
+    },
+    supportData.nutrition.length > 0 && {
+      label: formatMessage(messages.nutrition),
+      content: (
+        <NutritionTable
+          data={supportData.nutrition}
+          footnote={formatMessage(messages['nutritionDisclaimer'])}
+          link="https://island.is/greidsluthatttaka-vegna-naeringar-og-serfaedis"
```
Comment (thordurhhh) [inline libs/service-portal/health/src/screens/AidsAndNutrition/AidsAndNutrition.tsx:55]: Same as previous comment regarding links in CMS

File: libs/service-portal/health/src/screens/HealthOverview/HealthOverview.tsx [hunk 5]
```diff
@@ -2,51 +2,55 @@ import React from 'react'
...
+  //       defaultMessage: 'Fara á heilsuveru',
+  //     })}
+  //     institutionTitle={m.digitalIceland}
+  //     institutionSubtitle={m.incoming}
+  //     institutionDescription={m.visionSegment}
+  //     institutionHref="https://island.is/stafraent-island"
+  //     institutionLinkTitle={m.timelineLinkTitle}
+  //     figure="/assets/images/bench.svg"
+  //   />
+  // )
```
Comment (thordurhhh) [inline libs/service-portal/health/src/screens/HealthOverview/HealthOverview.tsx:53]: Remove?

File: libs/service-portal/health/src/components/ExpiringTable/ExpiringExpandedTableRow.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                  position: 'relative',
+                }}
+              >
+                <Text variant="medium">{val}</Text>
+              </T.Data>
+            ))}
+          </T.Row>
+        </T.Body>
+      </T.Table>
+              */}
```
Comment (thordurhhh) [inline libs/service-portal/health/src/components/ExpiringTable/ExpiringExpandedTableRow.tsx:63]: Remove?

**Review by thordurhhh — CHANGES_REQUESTED** [review]

File: libs/service-portal/health/src/components/FootNote.tsx/helper.tsx (hunk 1 above)
Comment (thorkellmani) [reply libs/service-portal/health/src/components/FootNote.tsx/helper.tsx:39]: Yeah, will remove

File: SUMMARY.md [hunk 7]
```diff
@@ -193,7 +193,7 @@
   - [Finance](libs/clients/finance/README.md)
   - [Firearm License](libs/clients/firearm-license/README.md)
   - [Clients Fishing License](libs/clients/fishing-license/README.md)
-  - [Health Insurance](libs/clients/health-insurance-v2/README.md)
+    - [Health Insurance](libs/clients/icelandic-health-insurance/health-insurance/README.md)
```
Comment (thordurhhh) [inline SUMMARY.md:196]: Screenshot 2023-05-23 at 11 00 13

👁️

Comment (thorkellmani) [reply SUMMARY.md]: fixed

File: libs/service-portal/health/src/components/ExpiringTable/ExpiringTable.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        <T.Table>
+          {header}
+          <T.Body>{children}</T.Body>
+        </T.Table>
+      </Box>{' '}
+      <Box paddingTop={4}>
+        <Text variant="small" paddingBottom={2}>
+          {footnote}
+        </Text>
+        {}
```
Comment (thordurhhh) [inline libs/service-portal/health/src/components/ExpiringTable/ExpiringTable.tsx:32]: 🤔

**Review by thordurhhh — APPROVED** [review]
Nice! 🙌

File: libs/api/domains/rights-portal/src/lib/api-domains-rights-portal.resolver.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+@Audit({ namespace: '@island.is/api/rights-portal' })
+export class RightsPortalResolver {
+  constructor(private readonly rightsPortalService: RightsPortalService) {}
+
+  @Scopes(ApiScope.health)
+  @Query(() => [Therapies], { nullable: true })
+  @Audit()
+  getRightsPortalTherapies(@CurrentUser() user: User) {
+    return this.rightsPortalService.getTherapies(user)
+  }
```
Comment (eirikurn) [inline libs/api/domains/rights-portal/src/lib/api-domains-rights-portal.resolver.ts:27]: All resolvers and mutations should follow our [naming convention](https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions). E.g.

```suggestion
  @Scopes(ApiScope.health)
  @Query(() => [Therapies], { name: 'rightsPortalTherapies', nullable: true })
  @Audit()
  getRightsPortalTherapies(@CurrentUser() user: User) {
    return this.rightsPortalService.getTherapies(user)
  }
```

File: libs/clients/icelandic-health-insurance/rights-portal/src/lib/rightsPortalClient.config.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+})
+
+export const RightsPortalClientConfig = defineConfig<z.infer<typeof schema>>({
+  name: 'RightsPortalConfig',
+  schema,
+  load: (env) => ({
+    xRoadServicePath: env.required(
+      'XROAD_HEALTH_INSURANCE_MY_PAGES_PATH',
+      'IS-DEV/GOV/10007/SJUKRA-Protected/minarsidur',
+    ),
```
Comment (eirikurn) [inline libs/clients/icelandic-health-insurance/rights-portal/src/lib/rightsPortalClient.config.ts:19]: Does this environment variable exist in the api infra file for all envs?

File: libs/api/domains/rights-portal/src/lib/api-domains-rights-portal.service.ts [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    })
+    throw new ApolloError('Failed to resolve request', error.status)
+  }
+
+  private handle4xx(error: any, detail?: string): ApolloError | null {
+    if (error.status === 403 || error.status === 404) {
+      return null
+    }
+    return this.handleError(error, detail)
+  }
```
Comment (eirikurn) [inline libs/api/domains/rights-portal/src/lib/api-domains-rights-portal.service.ts:37]: Maybe you copied this from elsewhere but we should start moving away from this error handling pattern. Enhanced Fetch includes standardized error logging, and unexpected errors should not be ignored or wrapped with ApolloError.

Also, the return type can be improved to simplify type handling elsewhere. `handleError` always throws so returns `never` and `handle4xx` only returns `null` (or throws an error).

File: libs/feature-flags/src/lib/features.ts [hunk 12]
```diff
@@ -44,6 +44,7 @@ export enum Features {
   servicePortalEducationGraduation = 'isServicePortalEducationGratuationModuleEnabled',
   sessionHistory = 'sessionHistory',
   servicePortalConsentModule = 'isServicePortalConsentModuleEnabled',
+  servicePortalHealthRightsModule = 'isServicePortalHealthRightsModuleEnabled',
```
Comment (eirikurn) [inline libs/feature-flags/src/lib/features.ts:47]: This feature should also be used to block access to the graphql resolvers. Can use the FeatureFlag nest module and decorator.

**Review by eirikurn — APPROVED** [review]
Looks good!

Most of my comments are not launch blockers. Main blockers are the infra configuration and maybe feature flagging the resolvers.

Otherwise approving to unblock. 🙌

**Review by eirikurn — APPROVED** [review]
Looks good!

Most of my comments are not launch blockers. Main blockers are the infra configuration and maybe feature flagging the resolvers.

Otherwise approving to unblock. 🙌

File: libs/clients/icelandic-health-insurance/rights-portal/src/lib/rightsPortalClient.config.ts (hunk 10 above)
Comment (thorkellmani) [reply libs/clients/icelandic-health-insurance/rights-portal/src/lib/rightsPortalClient.config.ts:19]: Its from the HealthInsurance xroad config

File: libs/api/domains/rights-portal/src/lib/api-domains-rights-portal.service.ts (hunk 11 above)
Comment (thorkellmani) [reply libs/api/domains/rights-portal/src/lib/api-domains-rights-portal.service.ts]: Yeah, simplified

File: libs/feature-flags/src/lib/features.ts (hunk 12 above)
Comment (thorkellmani) [reply libs/feature-flags/src/lib/features.ts:47]: ah yeah, will do

File: .github/CODEOWNERS [hunk 13]
```diff
@@ -144,6 +144,7 @@
 /apps/services/license-api/                                                                              @island-is/hugsmidjan
 /libs/regulations/                                                                                       @island-is/hugsmidjan
 /libs/portals/admin/regulations-admin/                                                                   @island-is/hugsmidjan
+/libs/clients/icelandic-health-insurance/rights-portal/                                                  @island-is/hugsmidjan
```
[the remark this answers is not in the document]
Comment (thorkellmani) [reply .github/CODEOWNERS:148]: oops

**Review by gudjong — APPROVED** [review]

**Pull request merged**